test/extended/operators/olm.go

```diff
@@ -70,4 +72,23 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 			}
 		})
 	}
+
+	// OCP-21082 - Implement packages API server and list packagemanifest info with namespace not NULL
+	// author: bandrade@redhat.com
+	g.It("Implement packages API server and list packagemanifest info with namespace not NULL", func() {
+		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "--all-namespaces", "-o=jsonpath={range .items[*]}{.metadata.name};{.status.catalogSourceNamespace}{\"\\n\"}").Output()
```

Comment:
This is checking the resource returned rather than the custom output columns. To satisfy [1670311](https://bugzilla.redhat.com/show_bug.cgi?id=1670311#c6), I think we want to list without the `-o=jsonpath` option, and parse the lines returned -- there should be four space delimited elements on each line, and the first two should match the namespace and name of an existing packagemanifest (except on line 0; the title line).

Reply:
@njhale I agree. If I consider that output may have less than 4 columns in one line if there is a regression(empty namespace) I have to check if it was the namespace column since the Catalog displayName is optional. So here is my suggestion:

```
g.It("Implement packages API server and list packagemanifest info with namespace not NULL", func() {
		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("packagemanifest", "--all-namespaces", "--no-headers").Output()
		if err != nil {
			e2e.Failf("Unable to get %s, error:%v", msg, err)
		}
		for _, packageManifest := range strings.Split(msg, "\n") {
			if string(msg[0]) == " " && len(strings.Fields(packageManifest)) < 4 {
				e2e.Failf("It should display a namespace for CSV: %s [ref:bz1670311]", packageManifest)
			}
		}
		o.Expect(err).NotTo(o.HaveOccurred())
	})
```
Please let me know if that looks good for you.

Reply:
Looks good, though I would expect the `o.Expect(err).NotTo(o.HaveOccurred())` assertion to  be immediately after the line setting the error value; i.e. `msg, err := ...`.

Reply:
Added the changes as oriented.